docs/recipes/probes.md

```diff
@@ -0,0 +1,172 @@
+# Probes in Tye
+
+Just like in a [Kubernetes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) deployment, you can define `liveness` and `readiness` ports in Tye, as part of the service.   
+
+Probes serve two main functions  
+
+* Keeping traffic away from replicas which are not ready to receive it.
+* Restarting replicas which are in a Bad/Unhealthy state.
+
+## Life cycle of a Replica
+
+Before we get to the definition of a probe and how it works, it's important to understand the life cycle of a replica.  
+
+Each replica in Tye can be in one of these three states  
```

Comment:
I'd try to emphasize that for tye deploy, Kubernetes is managing these states for started, healthy, and ready.